controllers/appgroup_controller.go

```diff
@@ -390,43 +381,121 @@ func (r *ApplicationGroupReconciler) handleRemediation(ctx context.Context, logr
 					err = r.List(ctx, &helmReleases, listOption)
 					if err != nil {
 						logr.Error(err, "failed to find generated HelmRelease instances")
-						return reconcile.Result{Requeue: false}, err
+						return reconcile.Result{Requeue: false}, nil
 					}
 
 					err = r.rollbackFailedHelmReleases(ctx, helmReleases.Items)
 					if err != nil {
 						logr.Error(err, "failed to rollback failed HelmRelease instances")
-						return reconcile.Result{Requeue: false}, err
+						return reconcile.Result{Requeue: false}, nil
 					}
 				}
 			}
 		}
 		// mark the object as requiring rollback so that we can rollback
 		// to the previous versions of all the applications in the ApplicationGroup
 		// using the last successful spec
-		g.Status.Phase = orkestrav1alpha1.Rollback
-		_ = r.Status().Update(ctx, &g)
-
-		return reconcile.Result{Requeue: true, RequeueAfter: requeueAfter}, nil
+		g.RollingBack()
+		_ = r.Status().Patch(ctx, &g, patch)
+		return reconcile.Result{RequeueAfter: requeueAfter}, nil
 	}
 	// Reverse and cleanup the workflow and associated helmreleases
+	g.RollingBack()
```

Comment:
@jonathan-innis Can we create a new condition like cleaning up since this isnt a rollback